src/components/AccordionPanel/AccordionPanelTrigger.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useCallback } from 'react'
+import styled, { css } from 'styled-components'
+
+import { AccordionPanelContext } from './AccordionPanel'
+import { AccordionPanelItemContext } from './AccordionPanelItem'
+import { Icon as IconComponent } from '../Icon'
+
+import { getIsInclude, mapToArray } from '../../libs/map'
+import { isTouchDevice } from '../../libs/ua'
+import { withTheme, InjectedProps } from '../../hocs/withTheme'
+
+type Props = {
+  children: React.ReactNode
+  className?: string
+}
+
+type MergedProps = Props & InjectedProps
+
+const AccordionPanelTriggerComponent: React.SFC<MergedProps> = ({
+  children,
+  className = '',
+  theme,
+}) => {
+  const { name } = useContext(AccordionPanelItemContext)
+  const { iconPosition, displayIcon, expandedItems, onClickTrigger, onClickProps } = useContext(
+    AccordionPanelContext,
+  )
+
+  const isExpanded = getIsInclude(expandedItems, name)
+  const expandedClassName = isExpanded ? 'expanded' : ''
+  const buttonClassNames = `${className} ${expandedClassName} ${iconPosition}`
+  const iconClassNames = `${expandedClassName} ${iconPosition}`
+
+  const caretIcon = <Icon className={iconClassNames} name="fa-caret-down" theme={theme} />
+
+  const handleClick = useCallback(() => {
+    onClickTrigger(name, !isExpanded)
+    if (onClickProps) onClickProps(mapToArray(expandedItems))
```

Comment:
[MUST] If `expandedItems` is passed to `onClickProps` here, it will not match the expected behavior. An array with `name` of the clicked element added to `expandedItems` must be passed to `onClickProps`.